### PR TITLE
Implemented assisted inject, further separation of concerns

### PR DESCRIPTION
**Pull request by JWGmeligMeyling**

### CODE REVIEW

**Comment by JeremybellEU** [comment]
Added some tests to increase coverage again. Looks good to me!

**Comment by JWGmeligMeyling** [comment]
LGTM

File: src/main/java/me/moodcat/backend/rooms/RoomInstanceFactory.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+/**
+ * Created by jwgmeligmeyling on 9-6-15.
+ */
+public interface RoomInstanceFactory {
+
+    /**
+     * Create a new {@link RoomInstance}.
+     *
+     * @param room
```
Comment (Azkle) [inline src/main/java/me/moodcat/backend/rooms/RoomInstanceFactory.java]: We miss javadoc, which is added in 134

**Comment by JWGmeligMeyling** [comment]
Closed because rebased in #134

**Pull request closed without merging**